asciidoc/courses/genai-mcp-build-custom-tools-python/modules/1-getting-started/lessons/1-mcp-python-sdk/lesson.adoc

```diff
@@ -0,0 +1,282 @@
+= Building MCP servers
+:type: lesson
+:order: 1
+
+In the link:/courses/genai-mcp-neo4j-tools/[Developing with Neo4j MCP Tools course], you learned the basics of the Model Context Protocol (MCP) and how to use the Neo4j Cypher MCP Server to enable AI agents to interact with Neo4j databases.
+
+In this course, you will learn how to build your own MCP servers with graph-backed tools and resources, creating a complete GraphRAG application.
+
+You will  learn to build these tools with the MCP Python SDK, and you will use the MCP Inspector to test your servers.
+
+[TIP]
+.Emerging standards
+====
+The Model Context Protocol is an emerging standard for connecting AI applications with tools and data sources. 
+As such the protocol is still evolving, and the features taught in this course are subject to change.
+
+You can follow link:https://modelcontextprotocol.io/[the Model Context Protocol documentation^] to stay up to date with the latest changes.
+====
+
+
+== Understanding the MCP Python SDK
+
+The MCP Python SDK is a comprehensive library that implements the full Model Context Protocol specification for Python applications.
+It provides both client and server implementations, making it easy to create MCP servers that expose custom functionality to AI agents.
+
+
+The SDK includes two main approaches for building servers:
+
+* **FastMCP**: A high-level, decorator-based approach that makes it simple to create servers quickly
+* **Low-level server**: A more flexible approach that gives you full control over the MCP protocol
+
+In this course, we will focus on the FastMCP approach.
+
+
+== Installing the MCP Python SDK
+
+To get started building MCP servers, you need to install the MCP Python SDK package.
+
+You can install it using pip:
+
+[source,bash]
+----
+pip install mcp
+----
+
+This package includes both the core MCP functionality and the FastMCP high-level interface.
+
+
+== Introducing FastMCP
+
+FastMCP is the high-level interface in the MCP Python SDK designed to make server development as simple as possible.  FastMCP is built on top of the FastAPI framework, and uses a decorator approach to defining MCP features.
+
+
+== Core MCP Features
+
+MCP servers expose three types of features to clients. Let's take a look at each one and when to use them.
```

Comment:
The content could be simplified, this act of reviewing them is implied through the content below.

```suggestion
MCP servers expose three types of features to clients.
```